src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -728,7 +735,16 @@ public URL getUrl() throws MalformedURLException {
         if (port == 0) {
             throw new IllegalStateException("This method must be called after calling #startJenkins.");
         }
-        return new URL("http", host, port, "/jenkins/");
+        return new URL(https ? "https" : "http", host, port, "/jenkins/");
+    }
+
+    public RealJenkinsRule https(File keyStoreFile, String keyStorePassword, SSLContext sslContext) {
```

Comment:
I do not think we need to deal with keystore passwords, do we? Presuming they are just test stores anyway.

Also it would be better to let the rule generate a keystore on the fly.